File: app/src/main/java/me/kirimin/mitsumine/domain/usecase/TopUseCase.java
package me.kirimin.mitsumine.domain.usecase;

import java.util.List;

import me.kirimin.mitsumine.data.database.AccountDAO;
import me.kirimin.mitsumine.data.database.FeedDAO;
import me.kirimin.mitsumine.data.database.KeywordDAO;
import me.kirimin.mitsumine.data.database.UserIdDAO;
import me.kirimin.mitsumine.model.Account;

public class TopUseCase {

    public List<String> getAdditionUsers() {
        return UserIdDAO.Companion.findAll();
    }

    public void deleteAdditionUser(String userId) {
        UserIdDAO.Companion.delete(userId);
    }

    public List<String> getAdditionKeywords() {
        return KeywordDAO.Companion.findAll();
    }

    public void deleteAdditionKeyword(String keyword) {
        KeywordDAO.Companion.delete(keyword);
    }

    public Account getAccount() {
        return AccountDAO.Companion.get();
    }

    public void deleteOldFeedData(){
        long threeDays = 1000 * 60 * 60 * 24 * 3;
        FeedDAO.Companion.deleteOldData(threeDays);
    }
}

File: app/src/main/java/me/kirimin/mitsumine/domain/usecase/TopUseCase.kt
package me.kirimin.mitsumine.domain.usecase

import me.kirimin.mitsumine.data.TopData
import me.kirimin.mitsumine.model.Account
import me.kirimin.mitsumine.model.enums.Type

open class TopUseCase {

    val data: TopData

    constructor(topData: TopData) {
        this.data = topData
    }

    open val additionUsers: List<String>
        get() = data.additionUsers

    open fun deleteAdditionUser(userId: String) {
        data.deleteAdditionUser(userId)
    }

    open val additionKeywords: List<String>
        get() = data.additionKeywords

    open fun deleteAdditionKeyword(keyword: String) {
        data.deleteAdditionKeyword(keyword)
    }

    open val account: Account?
        get() = data.account

    open fun deleteOldFeedData() {
        val threeDays = 1000 * 60 * 60 * 24 * 3.toLong()
        data.deleteOldFeedData(threeDays)
    }

    fun getTypeInt(type: Type): Int {
        return if (type == Type.HOT) 0 else 1
    }
}
